bugs/lp-1862702.txt

```diff
@@ -0,0 +1,1904 @@
+=== Begin SRU Template ===
+[Impact]
+Users will now be able to disable network config by passing the 'network-config=disabled'
+through a kernel cmdline argument. Right now, we must pass the following kernel command
+to disable networking: 'network-config={"config": "disabled"}', where '{"config": "disabled"}
+must be encoded in base64.
+
+[Test Case]
+
+```
+#!/bin/sh
+set -x
+# Manually deploy on a lxc container
+
+CMD_ARGUMENT="network-config=disabled"
+#This is a base64 string representing {"config": "disabled"}
+CMD_ARGUMENT_BASE64="network-config=eyJjb25maWciOiAiZGlzYWJsZWQifQ=="
+#This is a base64 string representing {"config": "disabled'}
+CMD_ARGUMENT_BASE64_INVALID="network-config=eyJjb25maWciOiAiZGlzYWJsZWQnfQ=="
+CMD_ARGUMENT_INVALID="network-config=foo"
+
+restart_cloudinit() {
+   name=$1
+
+   lxc exec $name -- DI_LOG=stderr /usr/lib/cloud-init/ds-identify --force
+   lxc exec $name -- cloud-init clean --logs
+   lxc exec $name -- cloud-init init --local
+   lxc exec $name -- cloud-init init > /dev/null
+   lxc exec $name -- cloud-init modules --mode=config
+   lxc exec $name -- cloud-init modules --mode=final
+   lxc exec $name -- cloud-init status --wait --long
+}
+
+reset_cmd_line_variable() {
```

Comment:
Nice functions for driving lxc config changes etc. It might be nice to think about putting some of these functions into sru-vars.template for use in future SRUs so we don't have to hunt down "the right way to do it"